### PR TITLE
Adds integration tests for alternat module

### PR DESCRIPTION
**Pull request by bwhaley**
Sets up [Terratest](https://terratest.gruntwork.io/) for Alternat. It takes a very long time to run because AWS takes ~30 minutes to delete ENIs associated with Lambda functions. It isn't great, but it's better than not having tests.

As an alternative, I considered using [`terraform test`](https://developer.hashicorp.com/terraform/language/tests). It would have been able to test the module, but not the user data setup script. Terratest is more powerful and flexible for more complex test cases, like logging in to the instance to validate the setup.

### CODE REVIEW

**Comment by bwhaley** [comment]
@arkadiyt I just ran in to this error:

```
keypair.go:41: UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:sts::110535878513:assumed-role/github-alternat-integration-test/GitHubActions is not authorized to perform: ec2:ImportKeyPair on resource: arn:aws:ec2:eu-central-1:110535878513:key-pair/99LiRk with an explicit deny in a service control policy.
```

Any chance of getting an exception, or is it impossible?

To explain what's happening here, Terratest to creates/imports a keypair that can be used to SSH to the instance and validate the userdata setup.

**Comment by arkadiyt** [comment]
@bwhaley your test is trying to use eu-central-1, try switching to us-east-1

**Comment by bwhaley** [comment]
Ahh, that's what the SCP is blocking. I had it picking an arbitrary region. Thanks for the reply.

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ NAT Gateways are dead. Long live NAT instances!
 
 Built and released with 💚 by <a href="https://chime.com"><img src="/assets/Chime_company_logo.png" alt="Chime Engineering" width="146"/></a>
 
-[![GitHub Actions](https://github.com/1debit/alternat/workflows/Build/badge.svg)](https://github.com/1debit/alternat/actions)
+[![GitHub Actions](https://github.com/1debit/alternat/workflows/Test/badge.svg)](https://github.com/1debit/alternat/actions)
```
Comment (inga) [inline README.md:7]: Wrong links. 😄

File: test/alternat_test.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+)
+
+func TestAlternat(t *testing.T) {
+	// os.Setenv("SKIP_setup", "true")	
+	// os.Setenv("SKIP_apply_vpc", "true")	
+	// os.Setenv("SKIP_apply_alternat_basic", "true")	
+	// os.Setenv("SKIP_validate_alternat_basic", "true")	
+	// os.Setenv("SKIP_validate_alternat_setup", "true")	
+	// os.Setenv("SKIP_validate_alternat_replace_route", "true")	
+	// os.Setenv("SKIP_cleanup", "true")
```
Comment (inga) [inline test/alternat_test.go:35]: Do you want to leave this here?

File: test/alternat_test.go [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+			return "", err
+	}
+
+	return aws.ToString(result.Reservations[0].Instances[0].PublicIpAddress), nil
+}
+
+func getThisPublicIp() (string, error) {
+	url := "https://api.ipify.org"
+	resp, err := http.Get(url)
+    if err != nil {
```
Comment (inga) [inline test/alternat_test.go:267]: Tabs or spaces... this file is mixed with both but I'd set it to just one.

**Review by inga — APPROVED** [review]
It would be nice to explain how to use the example in the README for new people wanting to use/contribute.

File: README.md (hunk 1 above)
Comment (bwhaley) [reply README.md]: Fixed in 8e30199

File: test/alternat_test.go (hunk 2 above)
Comment (bwhaley) [reply test/alternat_test.go]: Added a comment explaining this in 8e30199

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+name: Test
+
+on: push
```
Comment (ejholmes) [inline .github/workflows/test.yaml:3]: Anything we need to do to lock this down to not run from forks? I could see this potentially getting abused...

I _think_ the default is to now allow access to secrets from forks, but we may just wanna verify that if we haven't already.

**Review by ejholmes — APPROVED** [review]
Neato. I think if we ever pull out our shared modules into their own repo, this'd be a pretty good argument for doing it (making testing of them easier).

File: .github/workflows/test.yaml (hunk 4 above)
Comment (bwhaley) [reply .github/workflows/test.yaml:3]: CI processes will not run from forks. To trigger tests from a fork, somebody with write access to the repo has to push a commit.

Comment (bwhaley) [reply .github/workflows/test.yaml:3]: Ref: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

Comment (bwhaley) [reply .github/workflows/test.yaml:3]: Actually it looks like it's now possible to trigger it without pushing a commit. This must be a relatively new feature as it didn't used to work this way.

> If you are comfortable with running workflows on the pull request branch, return to the  Conversation tab, and under "Workflow(s) awaiting approval", click Approve and run.

Good call out.

**Pull request merged**